integration-tests/src/test/java/com/linkedin/kafka/clients/producer/LiKafkaProducerIntegrationTest.java

```diff
@@ -195,5 +209,77 @@ public void close() {
     assertEquals(1, messageCount);
   }
 
+  @Test
+  public void testCustomPartitioningOfLargeMessage() throws Exception {
+    Random random = new Random();
```

Comment:
Use fixed random number seed for test reproducibility.  Or rather... why even bother with this random partitions business.